src/ZM/Logger/TablePrinter.php

```diff
@@ -0,0 +1,253 @@
+            $k_len = mb_strwidth($k); // 获取 key 的宽度
+            $v_len = mb_strwidth($v); // 获取 value 的宽度
+            $len = $k_len + 2 + $v_len; // 计算需要的宽度
+            $valid_width = $this->border_width - 2; // 获取可用的宽度
+            if ($k_len + 5 > $this->border_width - 2) { // 这个参数的 key 过长了，没有你这么用的！！
+                continue;
+            }
+            while (true) {
+                if (!isset($line_data[$current_line])) { // 如果行是空的，先尝试放入一个参数
+                    if ($len > $valid_width) { // 需要的宽度超出一行，直接另起折行
+                        if ($this->row_overflow_hide) { // 如果开启了超出部分隐藏，则直接砍掉后面的东西，变成三个或四个点
+                            [$partial_v, $partial_v_len] = $this->getPartialValue($v, $valid_width - $k_len - 2);
+
+                            // 写入到数据中
+                            $line_data[$current_line] = [
+                                'used' => $valid_width - $k_len - 2 - $partial_v_len,
+                                'can_put_second' => false,
+                                'lines' => $k . ': ' .
+                                    ConsoleColor::apply([$this->value_color], $partial_v . str_pad('', $valid_width - $k_len - 2 - $partial_v_len, '.')),
+                            ];
+                            ++$current_line;
+                        // 下一个参数
+                        } else { // 没开隐藏就要拐弯输出
+                            $line_data[$current_line] = [
+                                'used' => $k_len + 2,
+                                'can_put_second' => false,
+                                'lines' => $k . ': ',
+                            ];
+                            do {
+                                [$partial_v, $partial_v_len, $next_offset] = $this->getPartialValue($v, $valid_width - $line_data[$current_line]['used'], 0);
+                                $v = mb_substr($v, $next_offset);
+                                $line_data[$current_line]['lines'] .= ConsoleColor::apply([$this->value_color], $partial_v);
+                                $line_data[$current_line]['used'] += $partial_v_len;
+                                ++$current_line;
+                                $line_data[$current_line] = [
+                                    'used' => 0,
+                                    'can_put_second' => false,
+                                    'lines' => '',
+                                ];
+                            } while ($v !== '');
+                            if ($line_data[$current_line]['used'] === 0) {
+                                unset($line_data[$current_line]);
+                            }
+                            // 下一个参数
+                        }
+                        break;
+                    }
+                    // 没超出一行，直接写
+                    $line_data[$current_line] = [
+                        'used' => $len,
+                        'can_put_second' => ($valid_width >= 57 && intval(floor($valid_width / 2)) - 2 + ($valid_width % 2) > $len),
```

Comment:
建议改用 `(int)`

Reply:
> If you call intval() with only one argument, it will be compiled to ZEND_CAST instruction by function zend_try_compile_special_func(), yielding identical opcode to that of (int). Thus, there’s no performance issue with intval() here.

Reply:
而且(int)更是需要 2 层括号，所以这里没考虑用